ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorSerializeRow.java

```diff
@@ -355,6 +379,13 @@ private void serializeStructWrite(
     serializeWrite.finishStruct();
   }
 
+  class VectorSerializeMapWriter extends VectorSerializeWriter {
+    @Override
+    void serialize(Object colInfo, Field field, int adjustedBatchIndex) throws IOException {
+      serializeMapWrite((MapColumnVector)colInfo, field, adjustedBatchIndex);
+    }
+  }
+
   private void serializeMapWrite(
```

Comment:
Nit: serializeMapWrite part of VectorSerializeMapWriter class?

Reply:
done